railties/test/secrets_test.rb

```diff
@@ -82,18 +80,17 @@ def teardown
   test "merging secrets with encrypted precedence" do
     run_secrets_generator do
       File.write("config/secrets.yml", <<-end_of_secrets)
-        test:
+        production:
           yeah_yeah: lets-go-walking-down-this-empty-street
       end_of_secrets
 
       Rails::Secrets.write(<<-end_of_secrets)
-        test:
+        production:
           yeah_yeah: lets-walk-in-the-cool-evening-light
       end_of_secrets
 
-      Rails.application.config.read_encrypted_secrets = true
-      Rails.application.instance_variable_set(:@secrets, nil) # Dance around caching 💃🕺
-      assert_equal "lets-walk-in-the-cool-evening-light", Rails.application.secrets.yeah_yeah
+      add_to_env_config("production", "config.read_encrypted_secrets = true")
+      assert_equal "lets-walk-in-the-cool-evening-light", rails("runner", "-e", "production", "print Rails.application.secrets.yeah_yeah")
```

Comment:
Can't we use `assert_equal "…", app.secrets.yeah_yeah` etc. or similar throughout instead of going through the runner?

If we can't, then let's at least use `print` throughout.

Reply:
I think the problem is that `run_secrets_generator` (improperly, IMO) boots the application inside the test process. Changing that to use `rails("g", "secrets")` (or whatever the command is) might simplify the rest of these, by deferring the application boot until after we've done whatever file-writing we wish.

Reply:
I think that this problem is to require `abstract_unit.rb`.
In `abstract_unit.rb`, since secrets is set in the subclass of `Rails::Application`, the secrets is fixedat that time. 
So, even if fix `run_secrets_generator`, I think that can not pass the test unless you modify `abstract_unit.rb`. 

Therefore, I think that it is appropriate to avoid using `abstract_unit.rb`, what about it? 
Also, using the `app` method makes it possible to directly reference `Rails.application`, so ydo not have to use runner. 

As described below.  

```diff 
-require "abstract_unit"
 require "isolation/abstract_unit"
 require "rails/generators"
 require "rails/generators/rails/encrypted_secrets/encrypted_secrets_generator"
@@ -20,12 +19,13 @@ def teardown
   test "setting read to false skips parsing" do
     run_secrets_generator do
       Rails::Secrets.write(<<-end_of_secrets)
-        test:
+        production:
           yeah_yeah: lets-walk-in-the-cool-evening-light
       end_of_secrets
 
-      Rails.application.config.read_encrypted_secrets = false
-      Rails.application.instance_variable_set(:@secrets, nil) # Dance around caching 💃🕺
+      add_to_env_config("production", "config.read_encrypted_secrets = false")
+      app("production")
+
       assert_not Rails.application.secrets.yeah_yeah
     end
   end
@@ -82,17 +82,18 @@ def teardown
   test "merging secrets with encrypted precedence" do
     run_secrets_generator do
       File.write("config/secrets.yml", <<-end_of_secrets)
-        test:
+        production:
           yeah_yeah: lets-go-walking-down-this-empty-street
       end_of_secrets
 
       Rails::Secrets.write(<<-end_of_secrets)
-        test:
+        production:
           yeah_yeah: lets-walk-in-the-cool-evening-light
       end_of_secrets
 
-      Rails.application.config.read_encrypted_secrets = true
-      Rails.application.instance_variable_set(:@secrets, nil) # Dance around caching 💃🕺
+      add_to_env_config("production", "config.read_encrypted_secrets = true")
+      app("production")
+
       assert_equal "lets-walk-in-the-cool-evening-light", Rails.application.secrets.yeah_yeah
     end
   end
@@ -108,7 +109,9 @@ def teardown
         config.dereferenced_secret = Rails.application.secrets.some_secret
       end_of_config
 
-      assert_equal "yeah yeah\n", rails("runner", "-e", "production", "puts Rails.application.config.dereferenced_secret", fork: false)
+      app("production")
+
+      assert_equal "yeah yeah", Rails.application.config.dereferenced_secret
     end
   end
 
@@ -141,7 +144,9 @@ def teardown
         assert_match(/production:\n\s*api_key: 00112233445566778899aabbccddeeff…\n/, File.read(tmp_path))
       end
 
-      assert_equal "00112233445566778899aabbccddeeff…\n", rails("runner", "-e", "production", "puts Rails.application.secrets.api_key", fork: false)
+      app("production")
+
+      assert_equal "00112233445566778899aabbccddeeff…", Rails.application.secrets.api_key
```

Reply:
> Therefore, I think that it is appropriate to avoid using abstract_unit.rb, what about it?

Yes, we definitely need to do that.

> Also, using the `app` method makes it possible to directly reference `Rails.application`.

Yeah, that looks great! 💯

Reply:
Yep, that's it!

Reply:
Thanks, I fixed!